src/WebJobs.Script.WebHost/Controllers/HostController.cs

```diff
@@ -286,8 +286,8 @@ public async Task<IActionResult> SetState([FromBody] string state)
         [RequiresRunningHost]
         public async Task<IActionResult> Warmup([FromServices] IScriptHostManager scriptHostManager)
         {
-            // Endpoint only for Windows Elastic Premium or Linux App Service plans
-            if (!(_environment.IsLinuxAppService() || _environment.IsWindowsElasticPremium()))
+            // Endpoint not supported for consumption plans.
+            if (_environment.IsLinuxConsumption() || _environment.IsWindowsConsumption())
```

Comment:
As part of this, we also need to update the IsWarmupRequest check here: https://github.com/Azure/azure-functions-host/blob/5b5323c12cc1976650225d81bae5458927db06d9/src/WebJobs.Script.WebHost/Middleware/HostWarmupMiddleware.cs#L73, and associated tests. That check is used to conditionally add the middleware.

It turns out that due to a bug, the middleware is ALWAYS added currently - because the code here https://github.com/Azure/azure-functions-host/blob/5b5323c12cc1976650225d81bae5458927db06d9/src/WebJobs.Script.WebHost/WebJobsApplicationBuilderExtension.cs#L57 is incorrectly doing `builder.Use` instead of `config.Use`. That should also be fixed. I actually have a fix for that in a set of perf changes I haven't sent for PR yet, but while you're updating this you could just fix that as well.